### PR TITLE
DOC/STY: Update Docstrings in test_meta

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #545, partial updates to #565 and #860
- Adds missing docstrings to unit tests for `test_meta`
- Removed duplicate test
- Renames several tests for clarity
- Improves docstring style for future compliance with #860

Note that many tests could be optimized or combined, this is out of scope here in the light of planned changes to metadata (ie, #767, #786, #789)

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

Using flake8-docstrings.
```
flake8 . --count --select=D,E,F,W --statistics
```

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.3

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/tests/test_meta.py [hunk 1]
```diff
@@ -819,25 +884,25 @@ def test_inequality_with_higher_order_meta3(self):
...
         meta['rpa'] = {'units': 'crazy', 'long_name': 'boo_whoo'}
         self.meta['higher'] = meta
 
     def test_assign_higher_order_meta_from_dict(self):
-        meta = pysat.Meta()
-        meta['dm'] = {'units': 'hey', 'long_name': 'boo'}
-        meta['rpa'] = {'units': 'crazy', 'long_name': 'boo_whoo'}
-        self.meta['higher'] = {'meta': meta}
-
-    def test_assign_higher_order_meta_from_dict_correct(self):
```
Comment (jklenzing) [inline pysat/tests/test_meta.py:833]: these two tests were identical, but only one had assert statements

File: pysat/tests/test_meta.py [hunk 2]
```diff
@@ -1067,6 +1162,8 @@ def test_contains_case_insensitive_w_ho(self):
         assert ('NEW21' not in self.meta)
 
     def test_get_variable_name_case_preservation(self):
+        """Test `meta.var_case_name` preserves the case (is case independent?).
+        """
```
Comment (jklenzing) [inline pysat/tests/test_meta.py:1166]: I may be misunderstanding this test (and those below), but are they for case independence (I still get the same answer regardless of case) or case preservation (the case I enter matters)?

File: pysat/tests/test_meta.py [hunk 3]
```diff
@@ -370,7 +376,7 @@ def test_inst_ho_data_assign_meta_different_labels(self):
                                                       'desc'] == 'are fun'
 
     def test_inst_assign_from_meta(self):
-        """Test Meta assignment form another meta object
+        """Test Meta assignment form another meta object.
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:379]:
```suggestion
        """Test Meta assignment from another meta object.
```

File: pysat/tests/test_meta.py (hunk 1 above)
Comment (rstoneback) [reply pysat/tests/test_meta.py:833]: I wish I could tell you what I was thinking when I wrote that one. A docstring probably would've helped. 🤷

File: pysat/tests/test_meta.py (hunk 2 above)
Comment (rstoneback) [reply pysat/tests/test_meta.py:1166]: Meta is supposed to preserve the case it sees first but respond to inputs insensitively. The var_case_name is the relevant method here. Setting meta up with 'NEW2' means that is what I should get from var_case_name as long as some case variant of 'new2' is provided to it.

It is checking preservation of output and independence of input.

File: pysat/tests/test_meta.py [hunk 4]
```diff
@@ -1076,6 +1173,8 @@ def test_get_variable_name_case_preservation(self):
         assert ('NEW2' == self.meta.var_case_name('NEW2'))
 
     def test_get_attribute_name_case_preservation(self):
+        """Test that getting attribute names preserves the case.
+        """
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:1177]: This test needs a check that the attribute name is stored as 'YoYoYO', based on the name. Currently tests preservation of case for input meta values, as well as tests case independence for accessing labels.

Comment (rstoneback) [reply pysat/tests/test_meta.py:1177]: Line 1198 has the goods.

Comment (rstoneback) [reply pysat/tests/test_meta.py:1177]: Perhaps, 'Test that meta labels and values preserve the case'

File: pysat/tests/test_meta.py [hunk 5]
```diff
@@ -1119,6 +1222,8 @@ def test_get_attribute_name_case_preservation_w_higher_order_2(self):
                 == 'YoYoYO')
 
     def test_get_attribute_name_case_preservation_w_ho_reverse_order(self):
+        """Test that getting attribute names preserves the case when reversed.
+        """
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:1226]: 'reversed assignment order' ? ... don't think there is room.

File: pysat/tests/test_meta.py [hunk 6]
```diff
@@ -1159,6 +1268,8 @@ def test_has_attr_name_case_preservation_w_higher_order(self):
 
     # check support on case preservation, but case insensitive
     def test_replace_meta_units_list_weird_case(self):
+        """Test that replacing meta units is case independence.
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:1271]:
```suggestion
        """Test that replacing meta units has case independence.
```
```suggestion
        """Test that replacing meta units is case independent.
```

File: pysat/tests/test_meta.py [hunk 7]
```diff
@@ -1222,7 +1336,8 @@ def test_transfer_attributes_to_instrument_strict_names(self):
                                                         strict_names=True)
 
     def test_transfer_attributes_to_instrument_strict_names_false(self):
-        """Test attr transfer with strict_names set to True"""
+        """Test attr transfer with strict_names set to True.
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:1339]:
```suggestion
        """Test attr transfer with strict_names set to False.
```

File: pysat/tests/test_meta.py [hunk 8]
```diff
@@ -1207,7 +1320,8 @@ def test_transfer_attributes_to_instrument_leading_(self):
         assert self.meta.__yo_yo == 'yo yo'
 
     def test_transfer_attributes_to_instrument_strict_names(self):
-        """Test attr transfer with strict_names set to True/False"""
+        """Test attr transfer with strict_names set to True/False.
```
Comment (rstoneback) [inline pysat/tests/test_meta.py:1323]:
```suggestion
        """Test attr transfer with strict_names set to True.
```

**Review by rstoneback — COMMENTED** [review]
Thanks @jklenzing!

File: pysat/tests/test_meta.py [hunk 9]
```diff
@@ -1067,6 +1162,8 @@ def test_contains_case_insensitive_w_ho(self):
         assert ('NEW21' not in self.meta)
 
     def test_get_variable_name_case_preservation(self):
+        """Test `meta.var_case_name` preserves the case (is case independent?).
```
Comment (jklenzing) [inline pysat/tests/test_meta.py:1165]: @rstoneback, how about this?
```suggestion
        """Test `meta.var_case_name` preserves the required output case.
```

File: pysat/tests/test_meta.py (hunk 5 above)
Comment (jklenzing) [reply pysat/tests/test_meta.py:1226]: Will the higher order tests be removed as part of #789?  If so, we may not need to spend too much time wordsmithing these.

File: pysat/tests/test_meta.py (hunk 7 above)
Comment (jklenzing) [reply pysat/tests/test_meta.py:1339]: I think I was going to combine these with parametrize, decided against it and forgot to update.

File: pysat/tests/test_meta.py [hunk 10]
```diff
@@ -126,14 +128,14 @@ def test_info_message_incorrect_input_meta_labels(self, input, caplog):
...
+        """Test the `Meta.__repr__` function.
         """
         self.out = self.meta.__repr__()
         assert isinstance(self.out, str)
         assert self.out.find('Meta(') >= 0
 
     def test_setting_nonpandas_metadata(self):
-        """ Test meta initialization with bad metadata
+        """Test meta initialization with bad metadata.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:139]:
```suggestion
        """Test meta initialization with bad metadata."""
```

File: pysat/tests/test_meta.py [hunk 11]
```diff
@@ -143,7 +145,7 @@ def test_setting_nonpandas_metadata(self):
                               (['units', 'long_name'], ['V', 'Longgggg']),
                               (['fill'], [-999])])
     def test_inst_data_assign_meta(self, labels, vals):
-        """ Test Meta initialization with data
+        """Test Meta initialization with data.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:149]:
```suggestion
        """Test Meta initialization with data."""
```

File: pysat/tests/test_meta.py [hunk 12]
```diff
@@ -169,7 +171,7 @@ def test_inst_data_assign_meta(self, labels, vals):
     @pytest.mark.parametrize("mlabel,slist", [("units", []),
                                               ("notes", ['A', 'B'])])
     def test_inst_data_assign_meta_string_list(self, mlabel, slist):
-        """ Test string assignment to meta with a list of strings
+        """Test string assignment to meta with a list of strings.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:175]:
```suggestion
        """Test string assignment to meta with a list of strings."""
```

File: pysat/tests/test_meta.py [hunk 13]
```diff
@@ -185,7 +187,7 @@ def test_inst_data_assign_meta_string_list(self, mlabel, slist):
         self.check_meta_settings()
 
     def test_init_labels_w_int_default(self):
-        """ Test MetaLabels initiation with an integer label type
+        """Test MetaLabels initiation with an integer label type.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:191]:
```suggestion
        """Test MetaLabels initiation with an integer label type."""
```

File: pysat/tests/test_meta.py [hunk 14]
```diff
@@ -215,14 +217,16 @@ def test_init_labels_w_int_default(self):
         self.check_meta_settings()
 
     def test_inst_data_assign_meta_empty_list(self):
+        """Test meta assignment from empty list.
+        """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:221]:
```suggestion
        """Test meta assignment from empty list."""
```

File: pysat/tests/test_meta.py [hunk 15]
```diff
@@ -215,14 +217,16 @@ def test_init_labels_w_int_default(self):
...
         self.testInst.load(2009, 1)
         self.testInst['help'] = {'data': self.testInst['mlt'],
                                  'units': [],
                                  'long_name': 'The Doors'}
         assert self.testInst.meta['help', 'units'] == ''
 
     def test_inst_data_assign_meta_then_data(self):
-        """ Test meta assignment when data updated after metadata
+        """Test meta assignment when data updated after metadata.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:230]:
```suggestion
        """Test meta assignment when data updated after metadata."""
```

File: pysat/tests/test_meta.py [hunk 16]
```diff
@@ -270,7 +276,7 @@ def test_inst_ho_data_assign_meta_default(self):
                 label)
 
     def test_inst_ho_data_assign_meta(self):
-        """ Test the assignemnt of custom higher order metadata
+        """Test the assignment of custom higher order metadata.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:280]:
```suggestion
        """Test the assignment of custom higher order metadata."""
```

File: pysat/tests/test_meta.py [hunk 17]
```diff
@@ -370,7 +376,7 @@ def test_inst_ho_data_assign_meta_different_labels(self):
                                                       'desc'] == 'are fun'
 
     def test_inst_assign_from_meta(self):
-        """Test Meta assignment form another meta object
+        """Test Meta assignment from another meta object.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:380]:
```suggestion
        """Test Meta assignment from another meta object."""
```

File: pysat/tests/test_meta.py [hunk 18]
```diff
@@ -463,7 +469,7 @@ def test_inst_assign_from_meta_w_ho_then_update(self):
         assert 'children' not in self.testInst.meta.data.columns
 
     def test_str_call_runs_long_standard(self):
-        """ Test long string output with custom meta data
+        """Test long string output with custom meta data.
         """
```
Comment (aburrell) [inline pysat/tests/test_meta.py:473]:
```suggestion
        """Test long string output with custom meta data."""
```

File: pysat/tests/test_meta.py (hunk 1 above)
Comment (aburrell) [reply pysat/tests/test_meta.py:833]: Let's remove the one without assert statements.

**Review by aburrell — COMMENTED** [review]
Other docstring cleaning puts the closing quotation marks on the same line if possible.  Also need to add return to end of tests.

**Comment by jklenzing** [comment]
Thanks!  Scrubbed again for style and returns.

**Review by rstoneback — APPROVED** [review]

**Comment by jklenzing** [comment]
Thanks @rstoneback.  @aburrell, any other comments?

**Review by aburrell — APPROVED** [review]

**Pull request merged**